src/security/reentrancyguard.cairo

```diff
@@ -14,14 +18,22 @@ mod ReentrancyGuard {
         const REENTRANT_CALL: felt252 = 'ReentrancyGuard: reentrant call';
     }
 
-    #[generate_trait]
-    impl InternalImpl of InternalTrait {
-        fn start(ref self: ContractState) {
+    trait InternalTrait<TContractState> {
+        fn start(ref self: TContractState);
+        fn end(ref self: TContractState);
+    }
+
+    impl InternalImpl<
+        TContractState, +HasComponent<TContractState>
+    > of InternalTrait<ComponentState<TContractState>> {
```

Comment:
```suggestion
    #[generate_trait]
    impl InternalImpl<
        TContractState, +HasComponent<TContractState>
    > of InternalTrait<TContractState> {
```

This should work (I hadn't figured it out when I wrote the trait in the SRC5 migration PR)